target/arm/cpu.c

```diff
@@ -136,6 +136,27 @@ static void cp_reg_reset(gpointer key, gpointer value, gpointer opaque)
     ARMCPRegInfo *ri = value;
     ARMCPU *cpu = opaque;
 
+#ifdef TARGET_CHERI
+    if (cpreg_field_is_cap(ri)) {
+        assert(ri->fieldoffset && "cpreg cap without offset?");
```

Comment:
Won't be hit yet, but an offset of zero is expected for any ARM_CP_SPECIAL that are also caps. Currently none are handled that way, but its allowed. Possibly the if (ri->type & ARM_CP_SPECIAL) should really be pulled before this if block to allow that to work? Or at least only throw this assert if not ARM_CP_SPECIAL?